src/uct/ib/base/ib_iface.c

```diff
@@ -347,52 +373,75 @@ unsigned uct_ib_iface_address_pack_flags(uct_ib_iface_t *iface)
 
 size_t uct_ib_iface_address_size(uct_ib_iface_t *iface)
 {
-    return uct_ib_address_size(&iface->gid_info.gid,
-                               uct_ib_iface_address_pack_flags(iface));
+    uct_ib_address_pack_params_t params;
+
+    params.flags     = uct_ib_iface_address_pack_flags(iface);
+    params.gid       = &iface->gid_info.gid;
+    params.roce_info = &iface->gid_info.roce_info;
+    return uct_ib_address_size(&params);
 }
 
-void uct_ib_iface_address_pack(uct_ib_iface_t *iface,
-                               uct_ib_address_t *ib_addr)
+void uct_ib_iface_address_pack(uct_ib_iface_t *iface, uct_ib_address_t *ib_addr)
 {
-    uct_ib_address_pack(&iface->gid_info.gid,
-                        uct_ib_iface_port_attr(iface)->lid,
-                        uct_ib_iface_address_pack_flags(iface),
-                        &iface->gid_info.roce_info, ib_addr);
+    uct_ib_address_pack_params_t params;
+
+    params.flags     = uct_ib_iface_address_pack_flags(iface);
+    params.gid       = &iface->gid_info.gid;
+    params.lid       = uct_ib_iface_port_attr(iface)->lid;
+    params.roce_info = &iface->gid_info.roce_info;
+    /* to suppress gcc 4.3.4 warning */
+    params.path_mtu  = 0;
+    params.gid_index = UINT8_MAX;
+    uct_ib_address_pack(&params, ib_addr);
 }
 
 void uct_ib_address_unpack(const uct_ib_address_t *ib_addr, uint16_t *lid,
-                           union ibv_gid *gid)
+                           union ibv_gid *gid, uint8_t *gid_index,
+                           enum ibv_mtu *path_mtu)
 {
     const void *ptr = ib_addr + 1;
 
-    *lid                      = 0;
+    *lid       = 0;
+    *gid_index = UINT8_MAX;
+    *path_mtu  = 0;
```

Comment:
maybe define 
UCT_IB_ADDRESS_INVALID_GID_INDEX  UINT8_MAX
UCT_IB_ADDRESS_INVALID_PATH_MTU  0

and use it in other places as well